### PR TITLE
change EBEforBSC to match EBE

### PR DESCRIPTION
**Pull request by fatemebagherii**

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -227,6 +236,7 @@ func ExecuteBlockEphemerallyForBSC(
 		return nil, fmt.Errorf("gas used by execution: %d, in header: %d", *usedGas, header.GasUsed)
 	}
 	if !vmConfig.NoReceipts {
+		bloom = newBlock.Bloom()
 		if newBlock.Bloom() != header.Bloom {
```
Comment (sudeepdino008) [inline core/blockchain.go]: statelessExec is used in other places in EBE...can you check those and ensure it's reflected here too

File: core/blockchain.go [hunk 2]
```diff
@@ -238,7 +248,19 @@ func ExecuteBlockEphemerallyForBSC(
 		return nil, fmt.Errorf("writing changesets for block %d failed: %w", header.Number.Uint64(), err)
 	}
 
-	return receipts, nil
+	execRs := &EphemeralExecResult{
+		TxRoot:      types.DeriveSha(includedTxs),
+		ReceiptRoot: receiptSha,
+		Bloom:       bloom,
+		// LogsHash:          rlpHash(blockLogs),
```
Comment (sudeepdino008) [inline core/blockchain.go]: why is LogsHash not present?

File: eth/stagedsync/stage_execute.go [hunk 3]
```diff
@@ -130,12 +130,17 @@ func executeBlock(
 	_, isPoSa := cfg.engine.(consensus.PoSA)
 	getHashFn := core.GetHashFn(block.Header(), getHeader)
 	if isPoSa {
-		receipts, err = core.ExecuteBlockEphemerallyForBSC(cfg.chainConfig, &vmConfig, getHashFn, cfg.engine, block, stateReader, stateWriter, epochReader{tx: tx}, chainReader{config: cfg.chainConfig, tx: tx, blockReader: cfg.blockReader}, contractHasTEVM)
+		getTracer := func(txIndex int, txHash ecom.Hash) (vm.Tracer, error) {
```
Comment (sudeepdino008) [inline eth/stagedsync/stage_execute.go:133]: check if you can use _ rather than using argument names here (txIndex,txHash) since they are not used

File: eth/stagedsync/stage_execute.go [hunk 4]
```diff
@@ -130,12 +130,17 @@ func executeBlock(
...
+		}
+		execRs, err = core.ExecuteBlockEphemerallyForBSC(cfg.chainConfig, &vmConfig, getHashFn, cfg.engine, block, stateReader, stateWriter, epochReader{tx: tx}, chainReader{config: cfg.chainConfig, tx: tx, blockReader: cfg.blockReader}, contractHasTEVM, true, getTracer)
+		receipts = execRs.Receipts
+		stateSyncReceipt = execRs.ReceiptForStorage
 	} else {
 		getTracer := func(txIndex int, txHash ecom.Hash) (vm.Tracer, error) {
 			return vm.NewStructLogger(&vm.LogConfig{}), nil
 		}
-		execRs, err = core.ExecuteBlockEphemerally(cfg.chainConfig, &vmConfig, getHashFn, cfg.engine, block, stateReader, stateWriter, epochReader{tx: tx}, chainReader{config: cfg.chainConfig, tx: tx, blockReader: cfg.blockReader}, contractHasTEVM, false, getTracer)
+		execRs, err = core.ExecuteBlockEphemerally(cfg.chainConfig, &vmConfig, getHashFn, cfg.engine, block, stateReader, stateWriter, epochReader{tx: tx}, chainReader{config: cfg.chainConfig, tx: tx, blockReader: cfg.blockReader}, contractHasTEVM, true, getTracer)
```
Comment (sudeepdino008) [inline eth/stagedsync/stage_execute.go]: why is this changed to true?

File: eth/stagedsync/stage_execute.go [hunk 5]
```diff
@@ -130,12 +130,17 @@ func executeBlock(
 	_, isPoSa := cfg.engine.(consensus.PoSA)
 	getHashFn := core.GetHashFn(block.Header(), getHeader)
 	if isPoSa {
-		receipts, err = core.ExecuteBlockEphemerallyForBSC(cfg.chainConfig, &vmConfig, getHashFn, cfg.engine, block, stateReader, stateWriter, epochReader{tx: tx}, chainReader{config: cfg.chainConfig, tx: tx, blockReader: cfg.blockReader}, contractHasTEVM)
+		getTracer := func(txIndex int, txHash ecom.Hash) (vm.Tracer, error) {
+			return vm.NewStructLogger(&vm.LogConfig{}), nil
+		}
+		execRs, err = core.ExecuteBlockEphemerallyForBSC(cfg.chainConfig, &vmConfig, getHashFn, cfg.engine, block, stateReader, stateWriter, epochReader{tx: tx}, chainReader{config: cfg.chainConfig, tx: tx, blockReader: cfg.blockReader}, contractHasTEVM, true, getTracer)
```
Comment (sudeepdino008) [inline eth/stagedsync/stage_execute.go]: statelessExec should be false here

File: core/blockchain.go [hunk 6]
```diff
@@ -161,31 +166,31 @@ func ExecuteBlockEphemerallyForBSC(
 		ibs.Prepare(tx.Hash(), block.Hash(), i)
 		writeTrace := false
 		if vmConfig.Debug && vmConfig.Tracer == nil {
-			vmConfig.Tracer = vm.NewStructLogger(&vm.LogConfig{})
+			tracer, err := getTracer(i, tx.Hash())
+			if err != nil {
+				panic(err)
```
Comment (sudeepdino008) [inline core/blockchain.go:171]: panic(err) indentation looks off (extra), can you fix that.

**Review by sudeepdino008 — CHANGES_REQUESTED** [review]

File: core/blockchain.go (hunk 2 above)
Comment (fatemebagherii) [reply core/blockchain.go]: because EBEforBSC didn't have anything related to it. So, I thought maybe it's kinda useless. But, I add it just in case

File: core/blockchain.go (hunk 6 above)
Comment (fatemebagherii) [reply core/blockchain.go:171]: are you sure? I don't see it

File: eth/stagedsync/stage_execute.go (hunk 3 above)
Comment (fatemebagherii) [reply eth/stagedsync/stage_execute.go:133]: I think there is no way

File: core/blockchain.go [hunk 7]
```diff
@@ -238,7 +248,43 @@ func ExecuteBlockEphemerallyForBSC(
 		return nil, fmt.Errorf("writing changesets for block %d failed: %w", header.Number.Uint64(), err)
 	}
 
-	return receipts, nil
+	// these parts were not present before. So, maybe we don't need them?
```
Comment (sudeepdino008) [inline core/blockchain.go]: this is not needed

**Review by sudeepdino008 — CHANGES_REQUESTED** [review]

File: core/blockchain.go (hunk 7 above)
Comment (fatemebagherii) [reply core/blockchain.go]: well, without them we can't have a proper ReceiptForStorage and LogsHash. Do you think that would be okay?

Comment (sudeepdino008) [reply core/blockchain.go]: that's okay

**Comment by sudeepdino008** [comment]
can you resolve the conflicts and comments? You can reference a corresponding commit in evm_t8n_2 branch as well

**Review by sudeepdino008 — APPROVED** [review]
